mantis-runtime/src/main/java/io/mantisrx/runtime/core/MantisStream.java

```diff
@@ -26,21 +27,97 @@
+     */
     <OUT> MantisStream<OUT> source(SourceFunction<OUT> sourceFunction);
 
+    /**
+     * Defines a sink operation that writes the elements in the stream to an
+     * external system, using the provided {@link SinkFunction}.
+     *
+     * This method defines a sink operation that writes the elements in the stream
+     * to an external system, using the provided {@link SinkFunction}. A sink
+     * operation is a terminal operation that does not produce a new stream, but
+     * instead writes the elements in the stream to an external system. The
+     * provided {@link SinkFunction} defines the logic to write the elements.
+     *
+     * @param sinkFunction a sink operator usually an SSE sink
+     * @return a {@link Config} object that captures the essence of {@link MantisJob}
+     */
     Config<T> sink(SinkFunction<T> sinkFunction);
 
+    /**
+     * Applies the provided {@link FilterFunction} to each element in the stream
+     * and returns all elements that match {@code filterFn.apply()}. The function is
+     * applied independently to each element in the stream.
+     *
+     * @param filterFn the function to apply on each element in the stream. Returns boolean
+     * @return a new {@link MantisStream}
+     */
     MantisStream<T> filter(FilterFunction<T> filterFn);
+
+    /**
+     * Applies the provided {@link MapFunction} to each element in the stream
+     * and returns a new stream consisting of result elements. The function is
+     * applied independently to each element in the stream.
+     *
+     * @param mapFn the function to apply to each element in the stream.
+     * @param <OUT> the type of the output elements in the resulting stream.
+     * @return a new {@link MantisStream}
+     */
     <OUT> MantisStream<OUT> map(MapFunction<T, OUT> mapFn);
+
+    /**
+     * Applies the provided {@link FlatMapFunction} to each element in the stream
+     * and returns a new stream consisting of result elements. The function is
+     * applied independently to each element in the stream. Compared to
+     * {@link MapFunction}, each element might produce zero, one, or more elements.
+     *
+     * @param flatMapFn the function to apply to each element in the stream.
+     * @param <OUT> the type of the output elements in the resulting stream.
+     * @return a new {@link MantisStream}
+     */
     <OUT> MantisStream<OUT> flatMap(FlatMapFunction<T, OUT> flatMapFn);
 
+    /**
+     * Operators are chained by default meaning processed in the same stage
+     * (worker-thread). Materialize breaks this chaining and all the pending
+     * operators are combined into a single scalar stage.
+     * Any subsequent operators will go into a new scalar stage [with chaining]
+     *
+     * @return a new {@link MantisStream} instance
+     */
     MantisStream<T> materialize();
 
+    /**
+     * Partitions the elements of the stream based on the key extracted by
```

Comment:
Following previous comment, we should call out important details such as local vs distributed.